crypto/store/store_lib.c

```diff
@@ -932,9 +932,16 @@ OSSL_STORE_CTX *OSSL_STORE_attach(BIO *bp, const char *scheme,
 
     OSSL_TRACE1(STORE, "Looking up scheme %s\n", scheme);
 #ifndef OPENSSL_NO_DEPRECATED_3_0
+    ERR_set_mark();
     if ((loader = ossl_store_get0_loader_int(scheme)) != NULL)
         loader_ctx = loader->attach(loader, bp, libctx, propq,
                                     ui_method, ui_data);
+    /*
+     * ossl_store_get0_loader_int will will raise an error if the loader for
+     * the scheme cannot be retrieved. This error is cleared as the code below
+     * will try to fetch the loader from the provider.
+     */
+    ERR_pop_to_mark();
```

Comment:
Ah!  Good thinking, but perhaps a little be coarse.  Have a look at `OSSL_STORE_open_with_libctx`, which does exactly this, but slightly differently to allow these errors to come through if there was no loader available at all.

Thank you, not having this error handling at all in here is an omission and does need fixing.

Reply:
Thanks, I'll take a closer look at how this is done in `OSSL_STORE_open_with_libctx`. This particular change is part of https://github.com/openssl/openssl/pull/12901 so I'll make the changes to that PR and in this PR. Sorry if this was confusing but without including that change the issue in this PR would not be noticed.

Reply:
No problem whatsoever, thanks for informing me.